### PR TITLE
Push Notifications

### PR DESCRIPTION
**Pull request by ovezovs**
Used Firebase Cloud Functions and Firebase Cloud Messaging (FCM) to send notification messages to a device's FCM token, which is a token unique for each device.

The current implementation of push notifications works in the following way:
1. **When users create a new account AND when they sign in**, a new FCM token is generated and stored in a document associated with the new user on Firestore.
2. **When users sign out**, the FCM token is removed from the user's document to avoid duplicate tokens in case a different user signs in from the same device.

**Design Summary:** Users receive notifications on the device they signed in the latest.

**NOTE:** This implementation is only temporary and should be improved in the later releases.

### CODE REVIEW

File: .idea/dictionaries/ovezovs.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<component name="ProjectDictionaryState">
+  <dictionary name="ovezovs">
```
Comment (hmanalai) [inline .idea/dictionaries/ovezovs.xml:2]: was the name "ovezovs" used for testing purposes? I think we should another name for it, more descriptive of the dictionary's functionality.

File: functions/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    let message = {
+      notification: {
+        title: title,
+        body: content
+      },
+      tokens: allTokens,
+    };
+
+    let response = await admin.messaging().sendMulticast(message);
+    console.log(response);
```
Comment (hmanalai) [inline functions/index.js:41]: Remove debugging statement

File: .idea/dictionaries/ovezovs.xml (hunk 1 above)
Comment (ovezovs) [reply .idea/dictionaries/ovezovs.xml]: I renamed the file to be more descriptive. Thanks!

File: functions/index.js (hunk 2 above)
Comment (ovezovs) [reply functions/index.js]: Fixed it!

**Pull request merged**